js/ui/windowManager.js

```diff
@@ -1481,23 +1560,23 @@ const WindowManager = new Lang.Class({
         Main.layoutManager.addChrome(button);
         let positionChangedId = window.connect('position-changed', Lang.bind(this, function(window) {
             let currentSession = this._currentEndlessCodingSessions.filter(function(session) {
-                return session.windowApp === window;
+                return session.windowApp.get_meta_window() === window;
             });
             if (currentSession.length === 0)
                 return;
-            let rect = currentSession[0].windowApp.get_frame_rect();
+            let rect = currentSession[0].windowApp.get_meta_window().get_frame_rect();
```

Comment:
Its up to you I guess. The distinction between the two can be annoying, but I guess if there's a way we can avoid long chains that'd be nice :)